web/packages/teleport/src/Instances/VersionsFilterPanel.tsx

```diff
@@ -0,0 +1,498 @@
+                        }}
+                        width="70px"
+                      />
+                      <Text color="text.muted" mx={1}>
+                        &
+                      </Text>
+                      <Input
+                        size="small"
+                        placeholder={currentVersion}
+                        value={customValue2}
+                        onChange={e => setCustomValue2(e.target.value)}
+                        disabled={selectedOption !== 'custom'}
+                        onFocus={() => {
+                          if (selectedOption !== 'custom') {
+                            handleOptionSelect('custom');
+                          }
+                        }}
+                        width="70px"
+                      />
+                    </>
+                  ) : (
+                    <Input
+                      ml={1}
+                      size="small"
+                      placeholder={currentVersion}
+                      value={customValue1}
+                      onChange={e => setCustomValue1(e.target.value)}
+                      disabled={selectedOption !== 'custom'}
+                      onFocus={() => {
+                        if (selectedOption !== 'custom') {
+                          handleOptionSelect('custom');
+                        }
+                      }}
+                    />
+                  )}
+                </Flex>
+                <ClearButton
+                  onClick={handleClearCustom}
+                  disabled={selectedOption !== 'custom'}
+                >
+                  <Icons.Cross size="small" />
+                </ClearButton>
+              </Flex>
+            </Box>
+          </Flex>
+        </FilterMenuItem>
+        <Divider />
+
+        <ActionButtonsContainer justifyContent="flex-start" p={3} gap={2}>
+          <ButtonPrimary size="small" onClick={handleApply}>
+            Apply Filters
+          </ButtonPrimary>
+          <ButtonSecondary
+            size="small"
+            css={`
+              background-color: transparent;
+            `}
+            onClick={handleClose}
+          >
+            Cancel
+          </ButtonSecondary>
+        </ActionButtonsContainer>
+      </Menu>
+    </Flex>
+  );
+}
+
+export function getMajorVersion(version: string): string {
+  const parsed = parse(version);
+  return `${parsed.major}.0.0`;
+}
+
+export function getMinorVersion(version: string): string {
+  const parsed = parse(version);
+  return `${parsed.major}.${parsed.minor}.0`;
+}
+
+export function getPreviousMajorVersion(version: string): string {
+  const majorNum = major(version);
+  return `${majorNum - 1}.0.0`;
+}
+
+export function getNextMajorVersion(version: string): string {
+  const majorNum = major(version);
+  return `${majorNum + 1}.0.0`;
+}
+
+// buildVersionPredicate returns the predicate query corresponding to a given version filter selection.
+export function buildVersionPredicate(
```

Comment:
Oh actually, one thing doesn't work for me: filtering by version. I have a v18.5.1 node agent connected to v19.0.0-dev local auth service. None of the built-in version filters return anything. The only thing that works is the custom condition, but only if I add the `v` prefix to a version.

It seems that those predicate functions expect them to be provided with the prefix maybe?